### PR TITLE
Fixes #194

### LINKED ISSUE #194: DataGenerator execute docstring

### PR DESCRIPTION
**Pull request by mpvanderschelling**
#194

### CODE REVIEW

File: src/f3dasm/_src/datageneration/datagenerator.py [hunk 1]
```diff
@@ -43,15 +44,32 @@ class DataGenerator:
...
     def execute(self, **kwargs) -> None:
-        """Function that calls the FEM simulator the pre-processing"""
-        raise NotImplementedError("No execute function implemented!")
+        """Interface function that handles the execution of the data generator
+
+        Raises
+        ------
+        NotImplementedError
+            If the function is not implemented by the user
+        """
```
Comment (BernardoFerreira) [inline src/f3dasm/_src/datageneration/datagenerator.py:63]: Please make clear in the method docstring that the ExperimentSample is accessed as an attribute of the DataGenerator class and should be updated in-place.

File: src/f3dasm/_src/datageneration/datagenerator.py [hunk 2]
```diff
@@ -43,15 +44,32 @@ class DataGenerator:
     """Base class for a data generator"""
```
Comment (BernardoFerreira) [inline src/f3dasm/_src/datageneration/datagenerator.py:44]: Interface should have a complete docstring explaining the sequence and expected functionality of the different methods involved in ._run().

It must be clear how the ExperimentSample object is stored and should be handled on each of these methods.

**Review by BernardoFerreira — CHANGES_REQUESTED** [review]
Please improve the documentation of the DataGenerator interface.

File: src/f3dasm/_src/datageneration/datagenerator.py (hunk 2 above)
Comment (mpvanderschelling) [reply src/f3dasm/_src/datageneration/datagenerator.py:44]: I've added more information about how the run function chains these methods together

File: src/f3dasm/_src/datageneration/datagenerator.py (hunk 1 above)
Comment (mpvanderschelling) [reply src/f3dasm/_src/datageneration/datagenerator.py:63]: I have added this information as a note to the docstring

File: src/f3dasm/_src/datageneration/datagenerator.py [hunk 3]
```diff
@@ -43,30 +44,80 @@ class DataGenerator:
...
         ...
 
     @time_and_log
     def _run(self, experiment_sample: ExperimentSample, **kwargs) -> ExperimentSample:
-        """Run the data generator
+        """
+        Run the data generator
+        This function chains the following methods together
+
+        * pre_process(); to combine the experiment_sample and the parameters
```
Comment (BernardoFerreira) [inline src/f3dasm/_src/datageneration/datagenerator.py:96]: Minor comment: Should these be ':' instead of ';'?

**Review by BernardoFerreira — APPROVED** [review]
Good job updating the docstrings of this interface! I approve the merging now.

Nonetheless, I still believe that the Sphinx documentation on this subject must be improved, for instance, when (some generic example) do you actually need the methods add_pre_processing() and add_post_processing?

There are some typos on the Sphinx docs as well: (1) `experimentdata.add_post_process(pre_process_function) ` on the post-processing section, (2) is the `execute()` meant to be inside the post-processing section?

**Comment by mpvanderschelling** [comment]
I created a new issue #209 to address further improvement of the docstrings. For now this issue will be closed!

**Pull request merged**